packages/ui/components/custom/date-range-picker.tsx

```diff
@@ -0,0 +1,515 @@
+  onChange?: (date: DateRange | undefined) => void;
+  minDate?: Date | null;
+  maxDate?: Date | null;
+  maxRangeInDays?: number;
+  showPresets?: boolean;
+  showWeekNumber?: boolean;
+}
+
+export function DateRangePicker({
+  className,
+  date,
+  onChange,
+  minDate = null,
+  maxDate = null,
+  maxRangeInDays,
+  showPresets = true,
+  showWeekNumber = true
+}: DateRangePickerProps) {
+  const [selectedRange, setSelectedRange] = React.useState<DateRange | undefined>(date);
+  const [currentYear, setCurrentYear] = React.useState(new Date().getFullYear());
+  const [open, setOpen] = React.useState(false);
+  const [calendarMonth, setCalendarMonth] = React.useState<Date>(date?.from || new Date());
+  const [presetsOpen, setPresetsOpen] = React.useState(false);
+  const [monthsOpen, setMonthsOpen] = React.useState(false);
+  const isMobile = useMediaQuery("(max-width: 768px)");
+
+  const today = React.useMemo(() => startOfToday(), []);
+
+  const months = React.useMemo(() => Array.from({ length: 12 }, (_, i) => format(new Date(2024, i, 1), "MMMM")), []);
+
+  const remainingRangeInDays = React.useMemo(() => {
+    if (!selectedRange?.from || !selectedRange?.to || !maxRangeInDays) {
+      return null;
+    }
+
+    const selectedRangeInDays = differenceInDays(selectedRange.to, selectedRange.from) + 1;
+
+    if (selectedRangeInDays >= maxRangeInDays) {
+      return 0;
+    }
+
+    return maxRangeInDays - selectedRangeInDays;
+  }, [selectedRange?.from, selectedRange?.to, maxRangeInDays]);
+
+  const effectiveMinDate = React.useMemo(() => {
+    if (remainingRangeInDays === null || !selectedRange?.from) {
+      return minDate;
+    }
+
+    const rangedMinDate = subDays(selectedRange.from, remainingRangeInDays);
+
+    return minDate ? max([minDate, rangedMinDate]) : rangedMinDate;
+  }, [minDate, selectedRange?.from, remainingRangeInDays]);
+
+  const effectiveMaxDate = React.useMemo(() => {
+    if (remainingRangeInDays === null || !selectedRange?.to) {
+      return maxDate;
+    }
+
+    const rangedMaxDate = addDays(selectedRange.to, remainingRangeInDays);
+
+    if (maxDate) {
+      return min([maxDate, rangedMaxDate]);
+    }
+
+    return rangedMaxDate;
+  }, [maxDate, selectedRange?.to, remainingRangeInDays]);
+
+  const presets = React.useMemo(
+    () => [
+      {
+        label: "Last 7 days",
+        getValue: () => ({
+          from: subDays(today, 6),
+          to: today
+        })
+      },
+      {
+        label: "Last 30 days",
+        getValue: () => ({
+          from: subDays(today, 29),
+          to: today
+        })
+      },
+      {
+        label: "Last 3 months",
+        getValue: () => ({
+          from: subMonths(today, 3),
+          to: today
+        })
+      },
+      {
+        label: "Last year",
+        getValue: () => ({
+          from: subYears(today, 1),
+          to: today
+        })
+      }
+    ],
+    [today]
+  );
+
+  const handlePresetSelect = React.useCallback(
```

Comment:
methods should state what they do, not when they're called. e.g.
```suggestion
  const selectRangeAndClosePresets = React.useCallback(
```
or anything better you may come up with

Reply:
Renamed all functions in this component